### PR TITLE
Plane: Add a failure timer for quadplane forward transitions

### PR DESCRIPTION
**Pull request by WickedShell**
This allows the user to specify a max transition time which if exceeded the vehicle will abort the forward transition and swap to `QLAND` and land in place. This is useful if your forward motor fails to spin for any reason. (In high winds you just watch the plane drift backwards helplessly burning battery).

`QLAND` was selected as it maps to the failsafe action I'm most interested in, we could have an extra parameter if people want a different failure action, but for a hands off autonomous system chilling in `QLOITER` wasn't a good choice so it's currently forcing `QLAND`.

This has currently only seen flight testing within SITL with non tailsitters.

### CODE REVIEW

**Comment by pompecukor** [comment]
I find this a great feature WickedShell. I know of a case where this would have saved the day.
You seem to bee on fire these days. Please keep up the good work.

File: ArduPlane/quadplane.cpp [hunk 1]
```diff
@@ -383,6 +383,15 @@ const AP_Param::GroupInfo QuadPlane::var_info2[] = {
     AP_SUBGROUPINFO(qautotune, "AUTOTUNE_",  6, QuadPlane, QAutoTune),
 #endif
 
+    // @Param: TRANS_FAIL_MS
+    // @DisplayName: Quadplane transition failure time
+    // @Description: Maximum time allowed for forward transitions, exceeding this time will cancel the transition and the aircraft will immediately change to QLAND. 0 for no limit.
+    // @Units: ms
```
Comment (tridge) [inline ArduPlane/quadplane.cpp]: I think we could make this in seconds, as an int16

Comment (WickedShell) [reply ArduPlane/quadplane.cpp]: Done.

**Comment by WickedShell** [comment]
@tridge I have done some testing with assist and it seems to be working as expected, but since I haven't ever done much with assist, that could use some extra testing/eyes. The other one that I haven't tested at all is tiltrotors (both as I don't have one and have never tried to SITL one before). I also added a tweak to reset the failure timer while not soft armed to prevent disarmed/immediately after arming failures.

**Comment by WickedShell** [comment]
Flight tested on a real aircraft, performs as expected.

**Pull request merged**